src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -1294,20 +1290,46 @@ internal static string ToString(ExecutionContext context, object obj, string sep
                     case TypeCode.UInt32:
                     case TypeCode.Int64:
                     case TypeCode.UInt64:
-                        return obj.ToString();
+                        str = obj.ToString();
+                        break;
                     case TypeCode.DateTime:
                         DateTime dt = (DateTime)obj;
-                        return dt.ToString(formatProvider);
+                        str = dt.ToString(formatProvider);
+                        break;
                     case TypeCode.Decimal:
                         Decimal dec = (Decimal)obj;
-                        return dec.ToString(formatProvider);
+                        str = dec.ToString(formatProvider);
+                        break;
                     case TypeCode.Double:
                         double dbl = (double)obj;
-                        return dbl.ToString(formatProvider);
-
+                        str = dbl.ToString(LanguagePrimitives.DoublePrecision, formatProvider);
+                        break;
                     case TypeCode.Single:
                         float sgl = (float)obj;
-                        return sgl.ToString(formatProvider);
+                        str = sgl.ToString(LanguagePrimitives.SinglePrecision, formatProvider);
+                        break;
+                    default:
+                        success = false;
+                        break;
+                }
+
+                return success;
+            }
+
+            PSObject mshObj = obj as PSObject;
+
+            #region plain object
+            if (mshObj == null)
+            {
+                if (obj == null)
+                {
+                    return string.Empty;
+                }
+
+                // Fast-track the primitive types...
```

Comment:
This comment doesn't really add new information beyond the method name

Reply:
Comments removed.